File: app/src/main/java/zebrostudio/wallr100/ui/buypro/BuyProContract.java
package zebrostudio.wallr100.ui.buypro;

import zebrostudio.wallr100.BasePresenter;
import zebrostudio.wallr100.BaseView;

public interface BuyProContract {

    interface BuyProView extends BaseView<BuyProPresenter>{

        void showPurchaseLoading();

        void showRestoreLoading();

        void showDataServiceOffError();
    }

    interface BuyProPresenter extends BasePresenter<BuyProView>{

        void onBuyProClick();

        void onRestorePurchaseClick();
    }

}

File: app/src/main/java/zebrostudio/wallr100/ui/buypro/BuyProContract.kt
package zebrostudio.wallr100.ui.buypro

import zebrostudio.wallr100.BasePresenter

interface BuyProContract {

  interface BuyProView

  interface BuyProPresenter : BasePresenter<BuyProView>{

    fun notifyBuyProClicked()
    fun notifyRestoreProClicked()

  }

}